qf/group_test.go

```diff
@@ -19,3 +19,349 @@ func TestGetUserSubset(t *testing.T) {
 		t.Errorf("GetUserSubset() mismatch (-wantSubset, +gotSubset):\n%s", diff)
 	}
 }
+
+func TestGroup_UserNames(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name  string
+		group *qf.Group
+		want  []string
+	}{
+		{
+			name:  "Empty group",
+			group: &qf.Group{},
+			want:  nil,
+		},
+		{
+			name: "Non empty group",
+			group: &qf.Group{
+				Users: []*qf.User{
+					{Login: "adityaa"},
+					{Login: "tootsy-tiger"},
+					{Login: "rhea"},
+				},
+			},
+			want: []string{"adityaa", "tootsy-tiger", "rhea"},
+		},
+		{
+			name:  "Nil group",
+			group: nil,
+			want:  nil,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
Just a quick comment about this line; since Go. 1.22, this is no longer necessary and can be removed from here and other tests below. For details see this [blog post](https://go.dev/blog/loopvar-preview).